src/herder/TxSetFrame.cpp

```diff
@@ -558,23 +262,23 @@ TxSetFrame::getTotalFees(LedgerHeader const& lh) const
 {
     ZoneScoped;
     auto baseFee = getBaseFee(lh);
-    return std::accumulate(mTransactions.begin(), mTransactions.end(),
+    return std::accumulate(mTxsInHashOrder.begin(), mTxsInHashOrder.end(),
                            int64_t(0),
                            [&](int64_t t, TransactionFrameBasePtr const& tx) {
                                return t + tx->getFee(lh, baseFee, true);
                            });
 }
 
 void
-TxSetFrame::toXDR(TransactionSet& txSet)
+TxSetFrame::toXDR(TransactionSet& txSet) const
 {
     ZoneScoped;
-    releaseAssert(std::is_sorted(mTransactions.begin(), mTransactions.end(),
+    releaseAssert(std::is_sorted(mTxsInHashOrder.begin(), mTxsInHashOrder.end(),
```

Comment:
this is probably also redundant now, given that it's true by construction.

Reply:
we can just keep the `releaseAssert` but check that the `mIsSorted` is true (you're just caching the result of `std::is_sorted`

Reply:
Updated.